pkg/vault/test/foundry/E2eErc4626Swaps.t.sol

```diff
@@ -346,17 +336,30 @@ contract E2eErc4626SwapsTest is BaseERC4626BufferTest {
         uint256 senderUsdcDelta = balancesBefore.balances.bobTokens[balancesBefore.usdcIdx] -
             balancesAfter.balances.bobTokens[balancesAfter.usdcIdx];
         uint256 vaultTotalUsdcBefore = balancesBefore.balances.vaultTokens[balancesBefore.usdcIdx] +
-            waUSDC.previewRedeem(balancesBefore.balances.vaultTokens[balancesBefore.waUsdcIdx]);
+            waUSDC.previewMint(balancesBefore.balances.vaultTokens[balancesBefore.waUsdcIdx]);
         uint256 vaultTotalUsdcAfter = balancesAfter.balances.vaultTokens[balancesAfter.usdcIdx] +
-            waUSDC.previewRedeem(balancesAfter.balances.vaultTokens[balancesAfter.waUsdcIdx]);
-        // TODO: check sign; vaultTotalUsdcAfter - vaultTotalUsdcBefore = senderUsdcDelta, solve for usdcAfter
-        // to prevent underflows
+            waUSDC.previewMint(balancesAfter.balances.vaultTokens[balancesAfter.waUsdcIdx]);
+
+        // `vaultTotalUsdcAfter - vaultTotalUsdcBefore = senderUsdcDelta`, solve for usdcAfter to prevent underflows.
         assertApproxEqAbs(
             vaultTotalUsdcAfter,
             senderUsdcDelta + vaultTotalUsdcBefore,
             MAX_ERROR,
             "Vault usdc/waUSDC balance is wrong"
         );
+
+        // DAI and USDC worths the same. So, we can sum the deltas for DAI and USDC in the vault, and compare with
+        // how much the user paid in fees. The user cannot have any benefit, but the vault and the user amount may
+        // differ because of rounding. Let's call the sum of vault deltas as sumDeltaVault, and the sum of sender
+        // deltas as sumDeltaSender, the test belows tests if
+        // `sumDeltaSender - 2 * MAX_ERROR < sumDeltaVault < sumDeltaSender`, it means, vault delta is smaller than
+        // sender delta (sender paid more than vault received), but the difference is small.
```

Comment:
Sounds good. 

How about also checking the wrapper rate before and after the operations? It doesn't really affect the vault, but wrapper rate increase basically explains this difference at the vault level (user pays more than what the vault receives, wrapper pockets the difference).

Reply:
Done!